File: 49.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
    	if(strs == NULL) return NULL;
    	hash_map<string, vector<string>> m;
    	for(auto s : strs) {
    		string t = s;
    		sort(s.begin(), s.end());
    		m[s].push_back(t);
    	}
    	vector<vector<string>> res;
    	auto it = m.begin();
    	while(it != m.end()) {
    		res.push_back(it -> second);
    		++it;
    	}
    	return res;
    }
};  

File: 49.py
from typing import List


class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        def str2str(s):
            chars = [0] * 26
            for c in s:
                chars[ord(c) - ord('a')] += 1
            return tuple(chars)

        dt = defaultdict(list)
        for s in strs:
            dt[str2str(s)].append(s)
        # ans = [l for l in dt.values()]
        # return ans
        return list(dt.values())


if __name__ == "__main__":
    s = Solution()
    print()
